### ISSUE TITLE
WSL Linux: Moby: mounting files is broken

### ISSUE THREAD

**Issue by mook-as** [rancher-sandbox/rancher-desktop#1307]
Trying to make a container using a file mount is broken.

## Steps to reproduce:
- On WSL (in a different Linux distro), create a file named `/tmp/my-secrets.json` and attempt to mount it in a container.
```sh
docker run --rm -t -i -v /tmp/my-secrets.json:/secrets.json:ro alpine ls -ld /secrets.json
```

## Expected results:
```
-rwx------    2 root     root            40 Jan 22 02:16 /secrets.json
```

## Actual results:
```
drwx------    2 root     root            40 Jan 22 02:16 /secrets.json
```

## Additional information:
This is a problem on the Linux docker proxy (which attempts to bind mount things into `/mnt/wsl/rancher-desktop/…` so that it can be seen by dockerd running in the rancher-desktop distribution):

```
ERRO[0003] could not perform bind mount                  bind=/mnt/wsl/rancher-desktop/run/docker-mounts/0b3e8acd-ab09-4b1c-9f87-94ff56a3a42d container=8db932291b5abaece0f03200ad1c0d4a3218f0056a8cf4825de50d48074af826 error="not a directory" target=/tmp/my-secrets.json
```

_Originally posted by @n0v1 in https://github.com/rancher-sandbox/rancher-desktop/issues/1186#issuecomment-1014356169_

**Comment by cmars** [comment]
I seem to be experiencing this issue in a fresh install of Rancher Desktop, was just trying it out on macos. Should this be fixed in 1.0.1?

**Comment by cmars** [comment]
Reason I suspect it's this issue, the way this Typescript app is failing when I try to run w/Rancher (using moby):

```
2022/02/04 11:17:26 running: docker run --rm -v /var/folders/xg/5fst00tj1m7gxf88yxqvrhfr0000gp/T/2864505601-input.json:/input.json -v /var/folders/xg/5fst00tj1m7gxf88yxqvrhfr0000gp/T/3122834436:/from/v3 -v /Users/cmars/Projects/golang-v3-sample-app/v3:/to/v3 snyk/sweater-comb:latest bulk-compare --input /input.json
Reading input file...
Bulk comparing


{
Bulk comparing


EISDIR: illegal operation on a directory, read
```

It works w/Docker Desktop and podman on Linux. I think it's likely to be that `input.json` mount...
